### PR TITLE
drivers: serial: stm32: Enable basic RX/TX FIFO

### PR DESCRIPTION
**Pull request by erwango**
On most STM32, a FIFO mode is available on U(S)ART blocks.
FIFO mode consists in a TX and RX FIFOs, each one 8 frames deep.

By enabling FIFO mode, both FIFOs will be used, using default threshold configuration.
No other change is made to the driver. 
In this case:
- TXE flag stands for TXFNF (TX empty become TXFIFO not full) 
- RXNE flag stands for RXFNE (RX not empty become RX Fifo not empty)
- same for matching interrupts

This has the following benefits:
In TX, FIFO allows to work in burst mode, easing scheduling of loaded applications. It also allows more reliable communication with UART devices sensitive to variation of inter-frames delays.
In RX, FIFO reduces overrun occurences.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

File: drivers/serial/uart_stm32.c [hunk 1]
```diff
@@ -1957,6 +1957,33 @@ static int uart_stm32_registers_configure(const struct device *dev)
 	}
 #endif
 
+#ifdef USART_CR1_FIFOEN
+	if (config->fifo_enable) {
+		LL_USART_EnableFIFO(config->usart);
```
Comment (reatreada) [inline drivers/serial/uart_stm32.c:1962]: Maybe it is worth setting `LL_USART_SetTXFIFOThreshold` and ` LL_USART_SetRXFIFOThreshold` explicitly or via DT?

Comment (GeorgeCGV) [inline drivers/serial/uart_stm32.c:1962]: Maybe it is worth setting `LL_USART_SetTXFIFOThreshold` and  `LL_USART_SetRXFIFOThreshold` explicitly or via DT?

**Review by dcpleung — APPROVED** [review]

**Review by GeorgeCGV — APPROVED** [review]

File: drivers/serial/uart_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/serial/uart_stm32.c:1962]: Actually, there is no code to handle thresholds, hence they have no effect.

File: dts/bindings/serial/st,stm32-uart-base.yaml [hunk 2]
```diff
@@ -86,3 +86,13 @@ properties:
...
+
+  fifo-enable:
+    type: boolean
+    description: |
+      Enables transmit and receive FIFO using default FIFO confugration (typically threasholds
+      set to 1/8).
+      In TX, FIFO allows to work in burst mode, easing scheduling of loaded applications. It also
+      allows more reliable communication with UART devices sensitive to variation of inter-frames
+      delays.
+      In RX, FIFO reduces overrun occurences.
```
Comment (GeorgeCGV) [inline dts/bindings/serial/st,stm32-uart-base.yaml:98]: Minor typos: `configuration`, `thresholds`, `occurrences`.

**Pull request merged**